src/main/java/htsjdk/tribble/IntervalList/IntervalListCodec.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2019, The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package htsjdk.tribble.IntervalList;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.FormatUtil;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.LineReader;
+import htsjdk.samtools.util.Log;
+import htsjdk.tribble.AsciiFeatureCodec;
+import htsjdk.tribble.TribbleException;
+import htsjdk.tribble.annotation.Strand;
+import htsjdk.tribble.readers.LineIterator;
+
+/**
+ * A tribble codec for IntervalLists.
+ *
+ * Also contains the parsing code for the non-tribble parsing of IntervalLists
+ */
+
+public class IntervalListCodec extends AsciiFeatureCodec<Interval> {
+
+    static final Log log = Log.getInstance(IntervalListCodec.class);
+
+    private SAMSequenceDictionary dictionary = null;
+
+    public IntervalListCodec() {
+        super(Interval.class);
+    }
+
+    public IntervalListCodec(final SAMSequenceDictionary dict) {
+        this();
```

Comment:
Normally when chaining constructors you want to chain from fewer arguments to more arguments, as the chaining lets you provide default values (or computed values) for arguments that can also be specified. In this case it would be better IMO to make the no-arg constructor call this constructor using `this(null);` and make this constructor call `super(Interval.class);`